packages/amplify-provider-awscloudformation/src/network/environment-info.ts

```diff
@@ -0,0 +1,151 @@
+import { JSONUtilities } from 'amplify-cli-core';
+import { EC2 } from 'aws-sdk';
+import { Netmask } from 'netmask';
+import configurationManager from '../configuration-manager';
+
+const SUBNETS = 3;
+type GetEnvironmentNetworkInfoParams = {
+  stackName: string;
+  vpcName: string;
+  vpcCidr: string;
+  subnetMask: number;
+  subnetsCount: number;
+};
+
+export async function getEnvironmentNetworkInfo(context, params: GetEnvironmentNetworkInfoParams) {
+  const { stackName, vpcName, vpcCidr, subnetsCount = SUBNETS, subnetMask } = params;
+
+  const [, vpcMask] = vpcCidr.split('/');
+
+  let cred = {};
+  try {
+    cred = await configurationManager.loadConfiguration(context);
+  } catch (e) {
+    // ignore missing config
+  }
+
+  const ec2 = new EC2({ ...cred });
+
+  const { AvailabilityZones } = await ec2.describeAvailabilityZones().promise();
+
+  if (subnetsCount > AvailabilityZones.length) {
+    const subnets = subnetsCount;
+    const AZs = AvailabilityZones.length;
+
+    throw new Error(
+      `The requested number of subnets exceeds the number of AZs for the region. ${JSONUtilities.stringify({ subnets, azs: AZs })}`,
+    );
+  }
+
+  const { Vpcs } = await ec2
+    .describeVpcs({
+      Filters: [{ Name: 'tag:Name', Values: [vpcName] }],
+    })
+    .promise();
+
+  if (Vpcs.length === 0) {
+    // we need a new one
+  }
+
+  if (Vpcs.length > 1) {
+    // which one to pick?
+    // btw, this shouldn't happen
+  }
+
+  const [vpc = {}] = Vpcs;
+
+  const { VpcId: vpcId } = vpc;
+
+  if (vpcId && !vpc.CidrBlock.endsWith(vpcMask)) {
+    throw new Error('Not the right mask'); // Should never happen
```

Comment:
Perhaps we should include the value of `vpcMask` to see what was the wrong value if happens.